Dockerfile

```diff
@@ -9,16 +9,15 @@
 #   > dotnet run Processor tcp://127.0.0.1:4004
 #   > dotnet run Client mykey set 42
 
-FROM microsoft/dotnet
+FROM microsoft/dotnet:sdk
 
-COPY . /sawtooth/sdk
+WORKDIR /sdk
+COPY . .
+RUN dotnet restore src/
 
-EXPOSE 4004 8008
+WORKDIR /core
+RUN git clone https://github.com/hyperledger/sawtooth-core.git
```

Comment:
Instead of cloning sawtooth-core, it would be better to copy the necessary protobuf files to this repository.

Primarily, we do this because we do not want a strict dependency between the SDK repositories and the sawtooth-core repository. Instead, we rely on binary compatibility of the protocol. So, for example, if we renamed a field in sawtooth-core proto files, which we can do based on how protobuf works without breaking ABI compatibility, we do not want SDKs to break because of a change in the generated source code which would result. Instead, we want the SDKs to change only when the changes are adopted explicitly.

Reply:
@vaporos Sounds good. If the protos are in the SDK repo, should we also keep a copy of the generated source code files in the repo? This would make the project build out of the box, while still leaving the option to the dev to generate them.
The docker file can then be repurposed to just generate a Nuget package or run the sample project.